python-lib/wordcloud_visualizer.py

```diff
@@ -0,0 +1,267 @@
+            languages = [self.language] * len(self.subcharts)
+        else:
+            languages = group_names
+
+        # Tokenize
+        docs = [self.tokenizer.tokenize_list(text, language)[0] for text, language in zip(texts, languages)]
+
+        return docs
+
+    @time_logging(log_message="Counting tokens")
+    def _count_tokens(self, docs: list) -> list:
+        """Private method to count tokens for each document in corpus
+        Args:
+            docs: list of spacy docs on which to count tokens
+        Returns:
+            List of tuples (subchart, counter) where subchart is the subchart the counter belongs to
+        """
+        counters = []
+        for doc in docs:
+            counter = Counter()
+            for token in doc:
+                counter[(token.text)] += 1  # Equivalently, token.lemma_
+            counters.append(counter)
+
+        if not self.subchart_column:
+            # Sum values with same keys
+            counts = Counter()
+            for d in counters:
+                counts.update(d)
+
+            counts = [("", dict(counts))]
+            return counts
+        else:
+            counts = list(zip(self.subcharts, counters))
+            # Aggregate counts by subchart
+            temp_count = {}
+            for subchart, count in counts:
+                if subchart in temp_count.keys():
+                    temp_count[subchart].update(count)
+                else:
+                    temp_count[subchart] = count
+            counts = list(temp_count.items())
+
+            # Remove subcharts emptied by aggregation
+            counts = [(subchart, count) for subchart, count in counts if count != {}]
+            return counts
+
+    def generate_wordclouds(self, counts: list):
+        """Public method to generate wordclouds and yield them as bytes-like objects
+        Args:
+            counts: list of tuples( subchart, counter) where subchart is the subchart the counter belongs to
+        Yields:
+            One tuple (bytes, filename) per subchart where bytes contains data from a wordcloud png file
+        """
+        if self.subchart_column:
+            for name, count in counts:
+                # Generate file name and chart title
+                output_file_name = pathvalidate.sanitize_filename(
+                    f"wordcloud_{self.subchart_column}_{name}.png"
+                ).lower()
+                wordcloud_title = output_file_name[:-4]
+                # Generate chart
+                if self.language_as_subchart:
+                    fig = self._generate_wordcloud(count, wordcloud_title, name)
+                else:
+                    fig = self._generate_wordcloud(count, wordcloud_title, self.language)
+                # Return chart
+                temp = BytesIO()
+                fig.savefig(temp, bbox_inches=self.bbox_inches, pad_inches=self.pad_inches, dpi=fig.dpi)
+                yield (temp, output_file_name)
+                plt.close()
```

Comment:
This line and line 253 are not executed since they come after a `yield`. Same behavior applies for `return` statements. You can remove the two lines.